### ISSUE TITLE
Maintaining Compatibility and Regular Updates for the Community App `files_mindmap`

### ISSUE THREAD

**Issue by nfebe** [nextcloud/files_mindmap#204]
Hello @ACTom 

I’m nfebe, a contributor and member of Nextcloud. Firstly, I want to thank you for all the hard work and dedication you’ve put into creating and maintaining `files_mindmap`. This app is highly valued by many in the Nextcloud community, and we’d love to support keeping it accessible and up to date.

Some users have recently reported compatibility issues with a couple of recent NC releases, which is impacting their experience with the app and Nextcloud as whole. It's understandable that managing updates for community-driven projects can be challenging, so here are a couple of suggestions we think would help:

- **Assistance with Releases**: If helpful, someone from the Nextcloud team could be assigned to handle release updates, provided they have the necessary permissions.
  
- **Transitioning to a NC managed repo**: If maintaining the app independently has become too time-intensive, we discused this at Nextcloud and we are totally okay with welcoming this repo as an official Nextcloud repository which would ensure more regular updates and streamline ongoing maintenance.

For reference, here are a few examples of compatibility issues reported by community members:
- [Issue #201](https://github.com/ACTom/files_mindmap/issues/201)
- [Issue #203](https://github.com/ACTom/files_mindmap/issues/203)
- [Issue #191](https://github.com/ACTom/files_mindmap/issues/191)
- [Issue #197](https://github.com/ACTom/files_mindmap/issues/197)
- [Issue #198](https://github.com/ACTom/files_mindmap/issues/198)

Please let us me your thoughts on these options or if there’s another approach that might be helpful. Thank you again for your work on this app.  I hope we can find a way to ensure it remains accessible to the Nextcloud community.

**Comment by ACTom** [comment]
Sorry, I was a bit busy a while ago and didn't take care of this project, it's changed a lot since NC28, I'm currently working on compatibility issues with this project and expect to release a new version in the near future.

**Comment by onny** [comment]
@ACTom I created a fork of your app over here and renamed it to files_mindmap2 https://github.com/onny/files_mindmap/

I'm looking forward to also add some improvements there and submit them here as pull requests too.

Would you be okay if we submit the fork files_mindmap2 to the Nextcloud app store until you've updated your version?

**Comment by ACTom** [comment]
@onny I've had a look at your code and there doesn't seem to be any substantial updates, the NC28+ compatibility issue is something I'd like to write in a new format rather than simply fixing the reported errors, I should have this done within a week.

**Comment by nfebe** [comment]
> Sorry, I was a bit busy a while ago and didn't take care of this project, it's changed a lot since NC28, I'm currently working on compatibility issues with this project and expect to release a new version in the near future.

Thanks for the update, do you have any timeline in mind? When can an updated be expected?

**Comment by dualizeo** [comment]
yeah any news on where can we get the next update? thanks 🙂

**Comment by ACTom** [comment]
It should be in the next few days. I think a version can be released on November 17 at the latest.

**Comment by AndyScherzinger** [comment]
Hi @ACTom ,

just wanted to circle back to the original discription of the issue description motivated by the community loving the app and looking for a v29 an v30 compatible version/release. Fon offered our help from Nextcloud (as a company) to support you in shouldering the work of maintaining and releasing the app. So I just wanted to get bck to that topic if you are fine to discuss it.

The app is loved by many but all the work is on you and we are happy to help, i.e. when moving it to the Nextcloud Github organization, contributions are simpler for members of that or, we can plug in the release automation we already have in place for many apps making it easy for a defined (and aligned) list of individuals having the ability to release the app with close to zero efforts to do so. Also being able to asily plugin into transifex for translations and using the bot we have to sync translation file changes.

So I was just wondering what your view is on that and if you would be open to such a change in sharing the ownership and work? To be clear you would of course still have all the right and permissions to the repository of course. It is not in any way meant to take anything away but rather to make your life easier and get releases out quicker in times when spare time to spent is low or one is sick or on vacation or not available for any other reason.

Looking forward to your feedback 🙏

**Comment by ACTom** [comment]
After the recent update, I will consider handing it over to community management. I hope it will be in a usable state when I hand it over.

**Comment by Yann-974** [comment]
@ACTom The new version as a XMas gift ?? :gift:

**Comment by beccon4** [comment]
> @onny I've had a look at your code and there doesn't seem to be any substantial updates, the NC28+ compatibility issue is something I'd like to write in a new format rather than simply fixing the reported errors, I should have this done within a week.

Why don't you work together to improve Mindmap? It obviously lacks developer's man power for ages so it can benefit from additional input.

**Comment by AndyScherzinger** [comment]
> After the recent update, I will consider handing it over to community management. I hope it will be in a usable state when I hand it over.

Hi @ACTom just wanted to check back with you on the matter of the 29/30 release and the community handover- Not wanting to create any pressure in any way. Just checking if there is anything we can do to help speed things up. We've all seen https://github.com/ACTom/files_mindmap/commit/def8c9862a156fb40e4cc9ea7995a76eafa75d05 from 2 month ago and on 25th February v31 of Nextcloud will be released. 
Also if it it simply too much work to complete at this point, please don't worry about that no need to improve the current state for a handover.

**Comment by poucz** [comment]
Any outlook on when it will be compatible with nextcoud 30?

**Comment by ACTom** [comment]
@poucz Just released [V0.0.32](https://github.com/ACTom/files_mindmap/releases/tag/v0.0.32), you can try it.

**Comment by ACTom** [comment]
@AndyScherzinger I Just finished syncing server-master and Nextcloud 31 has been overhauled? OCA.Viewer is gone.......

**Comment by poucz** [comment]
> @poucz Just released [V0.0.32](https://github.com/ACTom/files_mindmap/releases/tag/v0.0.32), you can try it.

Working, very well! Thank you!

**Comment by Uatschitchun** [comment]
> > After the recent update, I will consider handing it over to community management. I hope it will be in a usable state when I hand it over.
> 
> Hi [@ACTom](https://github.com/ACTom) just wanted to check back with you on the matter of the 29/30 release and the community handover- Not wanting to create any pressure in any way. Just checking if there is anything we can do to help speed things up. We've all seen [def8c98](https://github.com/ACTom/files_mindmap/commit/def8c9862a156fb40e4cc9ea7995a76eafa75d05) from 2 month ago and on 25th February v31 of Nextcloud will be released. Also if it it simply too much work to complete at this point, please don't worry about that no need to improve the current state for a handover.

Would it be possible to fork?

**Comment by AndyScherzinger** [comment]
@Uatschitchun Forking is always possible technically speaking but a agreed transfer would be the better/best option since it is the more friendly way and has the advantage that the whole reports with all it's data would transfer, else with a fork you loose all. Issue and PR and you are left to with purely the git side of things.

**Comment by ACTom** [comment]
@AndyScherzinger I'm sorry, I really don't have time to continue maintaining this project recently. If possible, I would like to transfer this project to Nextcloud Github Organisation and see what I need to do next?

**Comment by AndyScherzinger** [comment]
@Hi @ACTom thanks for reaching out and Nextcloud as a company and as a community is absolutely up for the task. In order to transfer you would need to transfer the GH repository see setting at https://github.com/ACTom/files_mindmap/transfer there you'd need to pick a user that has admin permissions on the Nextcloud Github organization, like me (I am the director of engineering at Nextcloud GmbH - the company, see i.e. https://nextcloud.com/team/ for "proof"):

<img width="219" alt="Image" src="https://github.com/user-attachments/assets/f4f5e66c-006c-4f61-9333-3a34ef022572" />

I will then be able to transfer it is to https://github.com/nextcloud/files_mindmap (only admins can transfer repositories to the organization). After that happened (you triggered the the transfer and I further transferred it) I will then grant you the permissions on that new home of your project, so you can still continue like you did here, but other can do yo as well.

---

**In short you just have to trigger the transfer**, everything else I can take care of via my colleagues at Nextcloud. Other that the repo transfer I suppose it would also make sense to do the very same with https://github.com/ACTom/kityminder-editor/transfer

**Looking forward to the transfer and thanks again for coming back to me on the matter and all you work for this magnificent app over the years** 💙 🎉 🏆

**Comment by ACTom** [comment]
@AndyScherzinger It says `AndyScherzinger/files_mindmap already exists`

**Comment by AndyScherzinger** [comment]
Sorry, sorry! My mistake... you should now be able, else just at a suffix like `1` to it

Could you please retry and sorry for the extra work 🙏

**Comment by ACTom** [comment]
@AndyScherzinger done.

**Comment by AndyScherzinger** [comment]
@ACTom I moved both now to the Nextcloud org (which this tracker as part of the repo is now too), see

* https://github.com/nextcloud/files_mindmap
* https://github.com/nextcloud/kityminder-editor

I also created 2 at 

* https://github.com/nextcloud/kity
* https://github.com/nextcloud/kityminder

Can't say if they are needed.

For the app I can now have somebody look into plugging the app into the translation platform for the translators community and setting up a automated packaging and publish to the app store.

I already granted you the maintainer permissions on the repositories 👍 

One thing you could also do on top (but I think we can also see how to do the transfer without your active help) would be to mark the app for transfer in the app store at https://apps.nextcloud.com/account/transfer-apps (in case you still have access to it, if not we'll figure out a way, so worries)

**Comment by AndyScherzinger** [comment]
Thanks a lot for all your help @ACTom and your work on the app 🎉 

90% of the transfer work is done, as posted above I can check with the developers at Nextcloud how we can do the last step alone or if you have some minutes to spare, you could also mark the app for transfer in the app store as per link above https://apps.nextcloud.com/account/transfer-apps 

Thanks again! 💙

**Comment by ACTom** [comment]
@AndyScherzinger Done, it's now Unlocked.

**Comment by AndyScherzinger** [comment]
Perfect, thanks @ACTom - I reached out to a knowledgable colleague and they will take care of the app store transfer part and are already looking into the translation service enablement.

So with this, we are now fully done with this ticket I think hence closing it.

Thanks @ACTom for everything and if you find the time you are always welcome and you have the system's permission as a maintainer set already and if you have time and motivation to work on mind-maps again you are always very very welcome 💙
